tests/integration/Core/Content/Category/SalesChannel/NavigationRouteTest.php

```diff
@@ -239,6 +246,226 @@ public function testInclude(): void
+            ->visibility($this->ids->get('sales-channel'), ProductVisibilityDefinition::VISIBILITY_ALL)
+            ->active(true);
+
+        $productBuilder->write($this->getContainer());
+
+        $this->getContainer()->get('category.repository')->update([
+            [
+                'id' => $this->ids->get('category4'),
+                'type' => CategoryDefinition::TYPE_LINK,
+                'linkType' => CategoryDefinition::LINK_TYPE_PRODUCT,
+                'internalLink' => $productId,
+            ],
+        ], Context::createDefaultContext());
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category4') && $category['linkType'] === CategoryDefinition::LINK_TYPE_PRODUCT) {
+                static::assertStringContainsString('/detail/' . $productId, $category['internalLink']);
+            }
+        }
+
+        $this->createSeoUrl(
+            'frontend.detail.page',
+            '/detail/' . $productId,
+            'custom-product-url',
+            $productId
+        );
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category4') && $category['linkType'] === CategoryDefinition::LINK_TYPE_PRODUCT) {
+                static::assertStringContainsString('/custom-product-url', $category['internalLink']);
+            }
+        }
+    }
+
+    public function testCategoryInternalLinkHasSeoUrl(): void
+    {
+        $this->getContainer()->get('category.repository')->update([
+            [
+                'id' => $this->ids->get('category3'),
+                'type' => CategoryDefinition::TYPE_LINK,
+                'linkType' => CategoryDefinition::LINK_TYPE_CATEGORY,
+                'internalLink' => $this->ids->get('category'),
+            ],
+        ], Context::createDefaultContext());
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category3') && $category['linkType'] === CategoryDefinition::LINK_TYPE_CATEGORY) {
+                static::assertNotEmpty($category['internalLink']);
+            }
+        }
+
+        $this->createSeoUrl(
+            'frontend.navigation.page',
+            '/navigation/' . $this->ids->get('category'),
+            'custom-category-url',
+            $this->ids->get('category')
+        );
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category3') && $category['linkType'] === CategoryDefinition::LINK_TYPE_CATEGORY) {
+                static::assertStringContainsString('/custom-category-url', $category['internalLink']);
+            }
+        }
+    }
+
+    public function testInternalLinkWithNullPlainUrl(): void
```

Comment:
remove this test case, it is testing an edge case that should not be tested in a integration test